### PR TITLE
Update provisioning schema and HOTH spec

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Update provisioning schema and HOTH specification to account for the changes in operators 
used for provisioning jobs, which aligns with the ATAT spec  version `0.3.0`. 
- Update provisioning job model and HOTH spec
- Update tests to reflect operator changes

Ticket: AT-7804

### CODE REVIEW

File: hoth_api.yaml [hunk 1]
```diff
@@ -299,11 +298,21 @@ components:
...
+      description: Operator data for provisioning resources in a CSP
+      type: object
+      properties:
+        email:
+          type: string
+          example: "user@mail.mil"
+        dod_id:
+          type: string
+        needs_reset:
+          type: boolean
```
Comment (kylelaker-ccpo) [inline hoth_api.yaml:315]: Are none of these required?

**Review by kylelaker-ccpo — COMMENTED** [review]
Great stuff! Just have a question to make sure we didn't missed a `required` field

File: hoth_api.yaml (hunk 1 above)
Comment (jasonburkert-ccpo) [reply hoth_api.yaml:315]: In the database schema I marked email and dod_id as mandatory, so perhaps do the same here

Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:315]: oh yeah, they are I'll fix that. Thanks! 👍

Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:315]: @jasonburkert-ccpo I made `needsReset` required too since version `0.3.0` of ATAT schema has it with a default of `false`.

Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:318]: Added in https://github.com/dod-ccpo/atat-web-api/pull/1098/commits/778c44a2da8872ab06991c8e176ab4112c552b96

**Review by kylelaker-ccpo — APPROVED** [review]

**Pull request merged**